lib/Parser/OglVersion.php

```diff
@@ -63,21 +63,44 @@ public function getGlslVersion() {
         return $this->glslVersion;
     }
 
-    // GL/GLSL extensions.
+    /**
+     * Add an extension, or merge it if it already exists.
+     *
+     * @param string $name Name of the extension.
+     * @param string $status Status of the extension.
+     * @param array $supportedDrivers List of drivers supported for this extension.
+     * @param \Mesamatrix\Git\Commit $commit The commit used by the parser.
+     *
+     * @return \Mesamatrix\OglExtension The new or existing extension.
+     */
     public function addExtension($name, $status, $supportedDrivers = array(), $commit = null) {
         $newExtension = new OglExtension($name, $status, $this->hints, $supportedDrivers);
         if ($extension = $this->getExtensionByName($name)) {
             $extension->incorporate($newExtension, $commit);
+            return $extension;
         }
         else {
             $this->extensions[] = $newExtension;
+            return $newExtension;
         }
     }
 
+    /**
+     * Get the list of all extensions.
+     *
+     * @return array All the extensions.
```

Comment:
If you use `OglExtension[]` instead of `array` static code checkers will sing praises for you :smile: